urlb879d9e865_ogotalski_elevator_tgz-pJ8-ElevatorTaskJ8/src/by/epam/lab/controller/Configuration.java

```diff
@@ -0,0 +1,119 @@
+package by.epam.lab.controller;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Properties;
+
+import by.epam.lab.Elevator;
+
+public class Configuration {
+
+	private static final String ANIMATION_BOOST_MUST_BE_LESS_THAN = "Animation boost must be less than ";
+	private static final String PROPERTIES_FILENAME = "config.properties";
+	private static final String PASSENGERS_NUMBER_MUST_BE_GREATER_THAN_0 = "Passengers number must be greater than 0";
+	private static final String STORIES_NUMBER_MUST_BE_GREATER_THAN_1 = "Stories number must be greater than 1";
+	private static final String ANIMATION_BOOST_MUST_BE_GREATER_THAN_0 = "Animation boost must be greater than 0";
+
+	private static final String ANIMATION_BOOST_PROPERTY_NAME = "animationBoost";
+	private static final String PASSENGERS_NUMBER_PROPERTY_NAME = "passengersNumber";
+	private static final String ELEVATOR_CAPACITY_PROPERTY_NAME = "elevatorCapacity";
+	private static final String STORIES_NUMBER_PROPERTY_NAME = "storiesNumber";
+
+	private static final int MIN_STORIES_NUMBER = 2;
+	private static final int MIN_PASSENGERS_NUMBER = 1;
+	private static final int MIN_ANIMATION_BOOST = 0;
+	private static final int SLEEP_MULTIPLIER = 25;
+	private static final int DEFAULT_SLEEP = 1000;
+	private static final int MAX_BOOST = DEFAULT_SLEEP / SLEEP_MULTIPLIER;
+	
+	private static Configuration configuration;
+
+	
+	public static Configuration getConfiguration() {
+		if (configuration == null)
+			configuration = new Configuration();
+		return configuration;
+	}
+
+	private int storiesNumber = MIN_STORIES_NUMBER;
+	private int elevatorCapacity = Elevator.MIN_ELEVATOR_COMPACITY;
+	private int passengersNumber = MIN_PASSENGERS_NUMBER;
+	private int animationBoost = MIN_ANIMATION_BOOST;
+
+	private Configuration() {
+		try {
+			init();
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+	}
+
+	public void init() throws IOException {
+		Properties prop = new Properties();
+
+		prop.load(new FileInputStream(PROPERTIES_FILENAME));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource created by the following constructor: *FileInputStream*. Execution paths that do not contain closure statements were detected. Ensure the object returned by *FileInputStream()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).